Kudu.Services/ServiceHookHandlers/VSOHandler.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Linq;
+using System.Web;
+using Kudu.Core.SourceControl;
+using Newtonsoft.Json.Linq;
+
+namespace Kudu.Services.ServiceHookHandlers
+{
+    /// <summary>
+    /// Default Servicehook Handler, uses VSO format.
+    /// </summary>
+    public class VSOHandler : ServiceHookHandlerBase
+    {
+        public override DeployAction TryParseDeploymentInfo(HttpRequestBase request, JObject payload, string targetBranch, out DeploymentInfo deploymentInfo)
+        {
+            deploymentInfo = null;
+
+            var publisherId = payload.Value<string>("publisherId");
+            if (String.Equals(publisherId, "tfs", StringComparison.OrdinalIgnoreCase))
+            {
+                GitDeploymentInfo gitDeploymentInfo = GetDeploymentInfo(request, payload, targetBranch);
+                deploymentInfo = gitDeploymentInfo;
+                return deploymentInfo == null || IsDeleteCommit(gitDeploymentInfo.NewRef) ? DeployAction.NoOp : DeployAction.ProcessDeployment;
+            }
+
+            return DeployAction.UnknownPayload;
+        }
+
+        protected virtual GitDeploymentInfo GetDeploymentInfo(HttpRequestBase request, JObject payload, string targetBranch)
+        {
+            // the rest must exist or NullRef.
+            // this allows us to identify the culprit handler
+            var sessionToken = payload.Value<JObject>("sessionToken");
+            var resource = payload.Value<JObject>("resource");
+            var refUpdates = resource.Value<JArray>("refUpdates");
+            var repository = resource.Value<JObject>("repository");
+
+            // The format of ref is refs/something/something else
+            // e.g. refs/head/master or refs/head/foo/bar
+            string branch = refUpdates.Last.Value<string>("name");
+            // Extract the name from refs/head/master notation.
+            int secondSlashIndex = branch.IndexOf('/', 5);
```

Comment:
good question.  I just copied the codes from [this](https://github.com/projectkudu/kudu/blob/master/Kudu.Services/ServiceHookHandlers/GitHubCompatHandler.cs#L46).